### PR TITLE
Feature 76 resource page

### PR DESCRIPTION
**Pull request by DexterFreeman**
Created resource page, with 3 new components/containers:

- StaffResources: Renders the cards and their labels. Along with the staff name (currently hard coded to be Staff 01, Staff 02 etc) Takes in cardsArray prop which is just the mockData. renderLabels is a prop that is either true or false which dynamically renders the labels above the card. 

- ResourcesOverview: This is the container for the Health resources and the Admin resources. Takes in one prop which is isAdmin which allows modularity so that the code knows whether it's rendering the admin resources or not. 

- StaffResourcesContainer: Container for all of the staff members within the "health resources" container. For future functionality all of the staff members will be passed in as a prop.  

There are two tests, one for ResourcesOverview and one for StaffResources. Both tests are to make sure the props function correctly. 

The page is styled for both mobile view and desktop view.

### CODE REVIEW

**Comment by DexterFreeman** [comment]
Just realised there's an error in my local host so gonna try fix that

**Comment by DexterFreeman** [comment]
Never mind, just needed to re run npm start after installing that new package :shipit:

File: src/components/StaffResources/StaffResources.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const StaffResources = ({staffName, renderLabels, cardsArray}) => {
+    const cardJSX = cardsArray.map((object, index) => {
+            console.log(object)
+            return <DataCard key={index} cardType="resource"
+            cardObject = {object}
+            />
+        })
+    
+
+    console.log(cardJSX)
```
Comment (TristanPere) [inline src/components/StaffResources/StaffResources.jsx:15]: Please remove white space and console.log

File: src/components/StaffResources/StaffResources.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            <p className="booking-container__label">Resource Name</p>
+            <p className="booking-container__label">Quantity Remaining</p>
+            <p className="booking-container__label">Cost per unit</p>
+            <p className="booking-container__label">Auto-Purchase</p>
+            <p className="booking-container__label">Auto-Purchase Level</p>
+        </div>}
+        <div className='staff-resources__cards'>
+       { cardJSX && cardJSX}
+       </div>
+
```
Comment (TristanPere) [inline src/components/StaffResources/StaffResources.jsx:30]: white space

File: src/components/StaffResources/StaffResources.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            cardObject = {object}
+            />
+        })
+    
+
+    console.log(cardJSX)
+  return (
+    <div className='staff-resources'>
+        <h1 className='staff-resources__staff-member'>{staffName}</h1>
+
```
Comment (TristanPere) [inline src/components/StaffResources/StaffResources.jsx:19]: white space

File: src/components/StaffResources/StaffResources.scss [hunk 4]
```diff
@@ -0,0 +1,57 @@
+@use "../../styles/abstracts/variables" as *;
+@use "../../styles/base/typography" as *;
+
+
```
Comment (TristanPere) [inline src/components/StaffResources/StaffResources.scss:4]: white space

File: src/components/StaffResources/StaffResources.scss [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    margin-bottom: 25px;
+    font-family: $font-modern;
+    
+    &__staff-member{
+        width: auto;
+        font-size: 14px;
+        font-weight: 400;
+        color: #95ACCB;
+        margin: 0 10%;
+
```
Comment (TristanPere) [inline src/components/StaffResources/StaffResources.scss:15]: white space

File: src/containers/ResourcesOverview/ResourcesOverview.jsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import "./ResourcesOverview.scss";
+import StaffResourcesContainer from '../StaffResourcesContainer/StaffResourcesContainer';
+import StaffResources from '../../components/StaffResources/StaffResources';
+import {resourceData} from "../../mockData.js"
+const ResourcesOverview = ({isAdmin}) => {
+  let resourceDataCopy =[...resourceData]
+  return (
+    
+    <div className='resources-overview'>
+
```
Comment (TristanPere) [inline src/containers/ResourcesOverview/ResourcesOverview.jsx:15]: white space

File: src/containers/ResourcesOverview/ResourcesOverview.jsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import listview from "../../assets/images/functional-icons/listview-icon.png";
+import SortIcon from "../../assets/images/functional-icons/sort-icon.png";
+import "./ResourcesOverview.scss";
+import StaffResourcesContainer from '../StaffResourcesContainer/StaffResourcesContainer';
+import StaffResources from '../../components/StaffResources/StaffResources';
+import {resourceData} from "../../mockData.js"
+const ResourcesOverview = ({isAdmin}) => {
+  let resourceDataCopy =[...resourceData]
+  return (
+
```
Comment (TristanPere) [inline src/containers/ResourcesOverview/ResourcesOverview.jsx:13]: whitespace

File: src/containers/ResourcesOverview/ResourcesOverview.jsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            className="resources-overview__icon"
+            src={gridview}
+            alt="Grid view icon"
+          />
+        </div>
+        <input
+          className="resources-overview__searchbar"
+          type="text"
+          name=""
+          id=""
```
Comment (TristanPere) [inline src/containers/ResourcesOverview/ResourcesOverview.jsx:35]: please remove if unassigned

File: src/containers/ResourcesOverview/ResourcesOverview.jsx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        </div>
+        </>
+      }
+      {isAdmin && 
+        <>
+             <h2 className="resources-overview__title">Admin</h2>     
+        </>
+      }
+      </div>
+
```
Comment (TristanPere) [inline src/containers/ResourcesOverview/ResourcesOverview.jsx:61]: whitespace

File: src/containers/ResourcesOverview/ResourcesOverview.jsx [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+      {isAdmin && 
+        <>
+             <h2 className="resources-overview__title">Admin</h2>     
+        </>
+      }
+      </div>
+          
+      {
+        isAdmin ? <><StaffResources cardsArray={resourceDataCopy}/></> : <StaffResourcesContainer/>
+
```
Comment (TristanPere) [inline src/containers/ResourcesOverview/ResourcesOverview.jsx:64]: white space

File: src/containers/ResourcesOverview/ResourcesOverview.jsx [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+      }
+
```
Comment (TristanPere) [inline src/containers/ResourcesOverview/ResourcesOverview.jsx:66]: whitespace

File: src/containers/ResourcesOverview/ResourcesOverview.jsx [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+      }
+      </div>
+          
+      {
+        isAdmin ? <><StaffResources cardsArray={resourceDataCopy}/></> : <StaffResourcesContainer/>
+
+      }
+      
+    </div>
+
```
Comment (TristanPere) [inline src/containers/ResourcesOverview/ResourcesOverview.jsx:68]: whitespace

File: src/containers/StaffResourcesContainer/StaffResourcesContainer.jsx [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+import './StaffResourcesContainer.scss'
+import StaffResources from '../../components/StaffResources/StaffResources'
+import {resourceData} from "../../mockData.js"
+const StaffResourcesContainer = () => {
+  let resourceDataCopy = [...resourceData]
+  return (
+    <div className='staff-resource-container'>
+        <StaffResources staffName={"Staff 01"} renderLabels={true} cardsArray={resourceDataCopy}/>
+        <StaffResources staffName={"Staff 02"} cardsArray={resourceDataCopy}/>
+
```
Comment (TristanPere) [inline src/containers/StaffResourcesContainer/StaffResourcesContainer.jsx:11]: whitespace

File: src/pages/Resources/Resources.scss [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+.resources-page{
+    display: flex;
+    min-height: 100vh;
+    height: 100%;
+    width: 100vw;
+    &__main-content{
+        width: 78vw;
+        height: 100%;
+        background-color: rgba(167, 167, 167, 0.06);
+
```
Comment (TristanPere) [inline src/pages/Resources/Resources.scss:14]: whitespace

File: src/pages/Resources/Resources.scss [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+        flex-direction: column;
+    }
+    &__header{
+        width: 100%;
+    }
+    &__cards-container{
+        display: flex;
+        flex-direction: column;
+    }
+
```
Comment (TristanPere) [inline src/pages/Resources/Resources.scss:25]: white space

File: src/pages/Resources/Resources.jsx [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+        <div className='resources-page__header'>
+        <PageHeader
+          heading="Resources"
+          text="Create"
+          headerType="headingButton"
+          />
+          </div>
+          <div className='resources-page__cards-container'>
+
+
```
Comment (TristanPere) [inline src/pages/Resources/Resources.jsx:21]: whitespace

**Review by TristanPere — CHANGES_REQUESTED** [review]
make changes discussed. otherwise looks great!

**Review by olirob93 — APPROVED** [review]

**Review by TristanPere — APPROVED** [review]

**Pull request merged**